Fix Warning C4706 - Using result of assignment as condition

A recent change has created a lot of compiler warnings:
```cpp
src/Configuration.cpp:219:18: warning: using the result of an assignment as a
      condition without parentheses [-Wparentheses]
                while (xmlNode = root->iterateChildren(xmlNode)) // warn...
                       ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is the result of many lines similar to the following:
```cpp
		while (xmlNode = root->iterateChildren(xmlNode)) // warning C4706: intended
```

Previously, these lines had an extra set of parenthesis which were silencing the warning:
```cpp
		while ((xmlNode = root->iterateChildren(xmlNode)))
```

----

It may be possible to provide a bit of support code to enable [range-based for loops](https://en.cppreference.com/w/cpp/language/range-for) to replace those `while` loops. In particular, the existence of `begin` and `end` methods can allow the compiler to generate a range-based for loop.

There are two ways to use `begin`/`end`. Either as member methods on a class, or as free floating methods which take an iterable parameter. If we wish to avoid modifying existing classes (perhaps from within the XML library), we can instead use the later form of free floating methods.

--- a/src/Configuration.cpp
+++ b/src/Configuration.cpp
@@ -200,8 +200,9 @@ bool Configuration::readConfig(const std::string& filePath)
 
 
 		// Start parsing through the Config.xml file.
-		XmlNode *xmlNode = nullptr;
-		while ((xmlNode = root->iterateChildren(xmlNode)))
+		for (auto xmlNode = root->iterateChildren(nullptr);
+			 xmlNode != nullptr;
+			 xmlNode = root->iterateChildren(xmlNode))
 		{
 			if (xmlNode->value() == "graphics") { parseGraphics(xmlNode); }
 			else if (xmlNode->value() == "audio") { parseAudio(xmlNode); }
@@ -349,12 +350,13 @@ void Configuration::parseOptions(void* _n)
 		return;
 	}
 
-	XmlNode *node = nullptr;
-	while ((node = element->iterateChildren(node)))
+	for (auto xmlNode = element->iterateChildren(nullptr);
+		 xmlNode != nullptr;
+		 xmlNode = element->iterateChildren(xmlNode))
 	{
-		if (node->value() == "option")
+		if (xmlNode->value() == "option")
 		{
-			XmlAttribute* attribute = node->toElement()->firstAttribute();
+			XmlAttribute* attribute = xmlNode->toElement()->firstAttribute();
 
 			std::string name, value;
 			while (attribute)
@@ -377,7 +379,7 @@ void Configuration::parseOptions(void* _n)
 
 			if (name.empty() || value.empty())
 			{
-				std::cout << "Invalid name/value pair in <option> tag in configuration file on row " << node->row() << ". This option will be ignored." << std::endl;
+				std::cout << "Invalid name/value pair in <option> tag in configuration file on row " << xmlNode->row() << ". This option will be ignored." << std::endl;
 			}
 			else
 			{
@@ -386,7 +388,7 @@ void Configuration::parseOptions(void* _n)
 		}
 		else
 		{
-			std::cout << "Unexpected tag '<" << node->value() << ">' found in configuration on row " << node->row() << "." << std::endl;
+			std::cout << "Unexpected tag '<" << xmlNode->value() << ">' found in configuration on row " << xmlNode->row() << "." << std::endl;
 		}
 	}
 }

